packages/counters.lua

```diff
@@ -52,91 +52,103 @@ SILE.formatCounter = function(options)
       display = "roman"
     end
 
-    local ok, res = pcall(function() return icu.format_number(options.value, display) end)
+    local ok, res = pcall(function() return icu.format_number(counter.value, display) end)
     if ok then return res end
   end
 
-  if (options.display == "roman") then return romanize(options.value):lower() end
-  if (options.display == "Roman") then return romanize(options.value) end
-  if (options.display == "alpha") then return alpha(options.value) end
-  return tostring(options.value)
+  if (counter.display == "roman") then return romanize(counter.value):lower() end
+  if (counter.display == "Roman") then return romanize(counter.value) end
+  if (counter.display == "alpha") then return alpha(counter.value) end
+  return tostring(counter.value)
 end
 
-local _init = function (c)
-  if not(SILE.scratch.counters[c]) then
-    SILE.scratch.counters[c] = { value= 0, display= "arabic" }
+local function getCounter(id)
+  local counter = SILE.scratch.counters[id]
+  if not counter then
+    counter = {}
+    SILE.scratch.counters[id] = { value= 0, display= "arabic", format = SILE.formatCounter }
   end
+  return counter
 end
 
-SILE.registerCommand("increment-counter", function (options,content)
-  local c = options.id; _init(c)
+SILE.registerCommand("increment-counter", function (options, _)
+  local counter = getCounter(options.id)
   if (options["set-to"]) then
-    SILE.scratch.counters[c].value = tonumber(options["set-to"])
+    counter.value = tonumber(options["set-to"])
   else
-    SILE.scratch.counters[c].value = SILE.scratch.counters[c].value + 1
+    counter.value = counter.value + 1
   end
-  if options.display then SILE.scratch.counters[c].display = options.display end
+  if options.display then counter.display = options.display end
 end, "Increments the counter named by the <id> option")
 
-SILE.registerCommand("set-counter", function (options, content)
-  local c = options.id; _init(c)
-  if options.value then SILE.scratch.counters[c].value = tonumber(options.value) end
-  if options.display then SILE.scratch.counters[c].display = options.display end
+SILE.registerCommand("set-counter", function (options, _)
```

Comment:
```suggestion
SILE.registerCommand("set-counter", function (options, content)
```

Please don't discard `content` here. Even though it doesn't get used by this command itself, we have ⓐ a stack trace that can inspect it and ⓑ a hook system that can do things with it. It is useful to always have all `SILE.registerCommand` registered functions have these two variables intact.